crates/driver/src/tests/cases/order_sorting.rs

```diff
@@ -0,0 +1,36 @@
+use crate::tests::{
+    setup,
+    setup::{ab_order, ab_pool, ab_solution},
+};
+
+/// Test that orders are sorted correctly before being sent to the solver:
+/// market orders come before limit orders, and orders that are more likely to
+/// fulfill come before orders that are less likely (according to token prices
+/// in ETH).
+#[tokio::test]
+#[ignore]
+async fn test() {
+    let test = setup()
+        .pool(ab_pool())
+        // Orders with better price ratios come first.
+        .order(
+            ab_order()
+                .reduce_amount(1000000000000000u128.into()),
+        )
+        .order(ab_order().rename("second order"))
+        // Limit orders come after market orders.
+        .order(
+            ab_order()
+                .rename("third order")
+                .limit()
+                .reduce_amount(1000000000000000u128.into()),
+        )
+        .order(ab_order().rename("fourth order").limit())
+        .solution(ab_solution())
+        .done()
+        .await;
+
+    // Only check that the solve endpoint can be called successfully, which means
+    // that the solver received the orders sorted.
```

Comment:
Not sure I get it. Can you point out where the code would throw an error if the orders were to be received in unsorted order?
Could be that I'm just not familiar enough with the test setup code.

Reply:
I think I found it. All good. 👌

Reply:
The test setup code is confusing. It could be better...